xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -324,11 +376,14 @@ private void updateChildState(
           childState = localityLbInfo.childHelper.currentChildState;
           childPicker = localityLbInfo.childHelper.currentChildPicker;
         }
-        overallState = aggregateState(overallState, childState);
 
-        if (READY == childState) {
-          childPickers.add(
-              new WeightedChildPicker(localityLbInfo.localityWeight, childPicker));
+        if (!localityMap.get(l).isDeactivated()) {
```

Comment:
The block above this if-statement doesn't have side-effect unless this if-condition is true. It's more readable to move that block into this if-block.

Reply:
Done.